### DISCUSSION TITLE
Load file based on metadata's label values for better geoip resolution

### DISCUSSION THREAD

**Discussion by arLevi** [coredns/coredns#6972] [Q&A] [answered]
I have the following example
```
example.com {
    view exampleshire {
      expr metadata('geoip/city/name') == 'Exampleshire'
    }
    geoip /opt/geoip2/db/GeoLite2-City.mmdb
    metadata
    file example.com.exampleshire-db
}
```

I'm trying to load different IP sets based on location, I was wondering if there is any simple way of doing "if" like statements in order to be able to resolve a domain based on its geoip information.

```
example.com {
    view exampleshire {
      expr metadata('geoip/continent/code') == 'Exampleshire'
    }
    geoip /opt/geoip2/db/GeoLite2-City.mmdb
    metadata
    file ./zones/example.com.{/geoip/continent/code}-db    <= taken from metadata(?)
}
```
And then i'll have files like
```
file ./zones/example.com.EU-db
file ./zones/example.com.US-db
etc...
```

== OR ==
Being able to rewrite the domain with different CNAME is also good for me. like:
```
example.com {
    view exampleshire {
      expr metadata('geoip/continent/code') == 'Exampleshire'
    }
    geoip /opt/geoip2/db/GeoLite2-City.mmdb
    metadata
    rewrite CNAME example.com {/geoip/continent/code}.example.com
}
```

It seems that `{/geoip/continent/code}` can only be served inside `log` format ..
how can I continue using the information inside the `metadata` to make decisions ?

**Comment by chrisohaver** [comment]
You should be able to do it with multiple example.com stanzas, each with different view criteria.

**Reply by arLevi** [reply]
`multiple example.com stanzas` multiple stanzas for many countries in the world will be a lot files on disk which i'm afraid might get a little spagehtti to me ...

I was wondering if there's an option to place a metadata variable inside the rule ?
Like: `file ./zones/example.com.{/geoip/continent/code}-db` or `rewrite CNAME example.com {/geoip/continent/code}.example.com` ?

**Reply by chrisohaver** [reply, marked as answer]
I don’t see how that would result in more files on disk than in your multiple _file_ plugin example. 

But perhaps the _template_ plugin is what you are looking for in regard to your dynamic CNAME example.

**Reply by arLevi** [reply]
Yes, `template` what what I needed, thanks !
